### PR TITLE
Allow use of CF ImportValue for httpApi id

### PR DESCRIPTION
**Pull request by fredericbarthelet**
Closes: #7791

### CODE REVIEW

File: lib/plugins/aws/info/getStackInfo.js [hunk 1]
```diff
@@ -28,10 +29,19 @@ module.exports = {
...
+    if (httpApiId) {
       sdkRequests.push(
-        this.provider
-          .request('ApiGatewayV2', 'getApi', { ApiId: this.serverless.service.provider.httpApi.id })
+        new BbPromise(resolve => {
+          if (httpApiId['Fn::ImportValue']) {
+            resolve(resolveCfImportValue(this.provider, httpApiId['Fn::ImportValue']));
+          }
+          resolve(httpApiId);
+        })
```
Comment (medikoo) [inline lib/plugins/aws/info/getStackInfo.js:41]: Let's construct it more optimally with:

```javascript
httpApiId['Fn::ImportValue'] ? resolveCfImportValue(this.provider, httpApiId['Fn::ImportValue']) : BbPromise.resolve(httpApiId) 
```

File: lib/plugins/aws/info/getStackInfo.js [hunk 2]
```diff
@@ -28,10 +29,19 @@ module.exports = {
...
-    if (this.serverless.service.provider.httpApi && this.serverless.service.provider.httpApi.id) {
+    const httpApiId =
+      this.serverless.service.provider.httpApi && this.serverless.service.provider.httpApi.id;
+    if (httpApiId) {
       sdkRequests.push(
-        this.provider
-          .request('ApiGatewayV2', 'getApi', { ApiId: this.serverless.service.provider.httpApi.id })
+        new BbPromise(resolve => {
+          if (httpApiId['Fn::ImportValue']) {
+            resolve(resolveCfImportValue(this.provider, httpApiId['Fn::ImportValue']));
```
Comment (medikoo) [inline lib/plugins/aws/info/getStackInfo.js:38]: If import is not resolved, I would crash with meaningful error

File: lib/plugins/aws/utils/resolveCfImportValue.test.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+const resolveCfImportValue = require('./resolveCfImportValue');
+
+describe('#resolveCfImportValue', () => {
+  let provider;
+  let providerStub;
+
+  beforeEach(() => {
+    const serverless = new Serverless();
+    provider = new AwsProvider(serverless, {});
+    providerStub = sinon.stub(provider, 'request');
```
Comment (medikoo) [inline lib/plugins/aws/utils/resolveCfImportValue.test.js:16]: Let's not use `Serverless` and `AwsProvider`, there's a lot of machinery in them which may introduce side effects.

I would simply inject plain `provider` object with `request` method that returns pre-prepared responses.

**Comment by fredericbarthelet** [comment]
@medikoo I updated the PR according to your comments. Let me know if it's fine now :)

File: lib/plugins/aws/info/getStackInfo.js [hunk 4]
```diff
@@ -28,10 +29,24 @@ module.exports = {
...
+          : BbPromise.resolve(httpApiId)
+        )
+          .then(id => {
+            if (!id) {
+              throw new this.serverless.classes.Error(
+                `Could not resolve provider.httpApi.id parameter. Configured value: ${JSON.stringify(
+                  httpApiId
+                )}`
+              );
+            }
```
Comment (medikoo) [inline lib/plugins/aws/info/getStackInfo.js:47]: Logically this should be attached to `resolveCfImportValue(this.provider, httpApiId['Fn::ImportValue']).then(..`

**Comment by fredericbarthelet** [comment]
PR updated @medikoo :)

**Pull request merged**